addons/mail/controllers/thread.py

```diff
@@ -109,8 +109,11 @@ def mail_message_post(self, thread_model, thread_id, post_data, context=None, sp
             post_data["body"] = Markup(post_data["body"])  # contains HTML such as @mentions
         new_partners = []
         if "partner_emails" in kwargs:
+            additional_values = kwargs.get("partner_additional_values", {})
+            for email in list(additional_values.keys()):  # Ensure normalized format (case: can be given name_addr format)
+                additional_values[email_normalize(email)] = additional_values.pop(email)
```

Comment:
`{email_normalize(email): values for email, values in kwargs.get("partner_additional_values", {})}` for the comprehension supremacy

Also avoids messing with the input params, and annoying "Iterating over the structure I'm modifying" kind of deals.